reflex/utils/prerequisites.py

```diff
@@ -281,31 +284,45 @@ def install_node():
         FileNotFoundError: if unzip or curl packages are not found.
         Exit: if installation failed
     """
+    # NVM is not supported on Windows.
     if IS_WINDOWS:
         console.print(
             f"[red]Node.js version {constants.NODE_VERSION} or higher is required to run Reflex."
         )
         raise typer.Exit()
 
-    # Only install if bun is not already installed.
-    console.log("Installing nvm...")
-
-    # Check if curl is installed
-    # TODO no need to shell out to curl
-    curl_path = path_ops.which("curl")
-    if curl_path is None:
-        raise FileNotFoundError("Reflex requires curl to be installed.")
-
     # Create the nvm directory and install.
-    path_ops.mkdir(constants.NVM_ROOT_PATH)
-    result = subprocess.run(constants.INSTALL_NVM, shell=True)
-
+    path_ops.mkdir(constants.NVM_DIR)
+
+    # Get the nvm install script.
+    response = httpx.get(constants.NVM_INSTALL_URL)
+    if response.status_code != httpx.codes.OK:
+        response.raise_for_status()
+
+    # Save the nvm install script to a temporary file.
+    script = tempfile.NamedTemporaryFile()
+    with open(script.name, "w") as f:
+        f.write(response.text)
+
+    # Run the nvm install script.
+    env = {
+        **os.environ,
+        "NVM_DIR": constants.NVM_DIR,
+    }
+    result = subprocess.run(["bash", f.name], env=env)
     if result.returncode != 0:
         raise typer.Exit(code=result.returncode)
 
-    console.log("Installing node...")
-    result = subprocess.run(constants.INSTALL_NODE, shell=True)
-
+    # Install node.
+    print("installing node")
+    result = subprocess.run(
+        [
+            "bash",
+            "-c",
+            f". {constants.NVM_DIR}/nvm.sh && nvm install {constants.NODE_VERSION}",
```

Comment:
Add comment that we have to do this "bash -c" business because we need to source in `. nvm.sh`?

Reply:
added a comment